src/cli.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import os
+from pathlib import Path
+from datetime import datetime
+
+from dotenv import load_dotenv, find_dotenv
+import click
+from tqdm.auto import tqdm
+from azure.core.exceptions import HttpResponseError
+from cpr_data_access.parser_models import PDFData, ParserOutput, BackendDocument
+
+from src.azure_pdf_parser import AzureApiWrapper
+from src.azure_pdf_parser.convert import (
+    extract_azure_api_response_paragraphs,
+    extract_azure_api_response_page_metadata,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+empty_backend_document = BackendDocument(
+    name="",
+    description="",
+    import_id="",
+    family_import_id="",
+    slug="",
+    publication_ts=datetime(1900, 1, 1),
+    source_url=None,
+    download_url=None,
+    type="",
+    source="",
+    category="",
+    geography="",
+    languages=[],
+    metadata={},
+)
+
+
+@click.command()
+@click.option(
+    "--pdf-dir",
+    help="Path to directory containing pdfs to process.",
+    required=True,
+    type=click.Path(exists=True, file_okay=False, path_type=Path),
+)
+@click.option(
+    "--output-dir",
+    help="Path to directory to write output JSONs to. Filenames and document IDs are be the filenames of the PDFs without extensions. Directory will be created if it doesn't exist.",
+    required=True,
+    type=click.Path(file_okay=False, path_type=Path),
+)
+def cli(pdf_dir: Path, output_dir: Path):
+    """Run Azure PDF parser on a directory of PDFs. Outputs 'blank' parser output jsons to `--output-dir`, with just document ID, document name, text block and page metadata information populated."""
+    load_dotenv(find_dotenv())
+
+    if not output_dir.exists():
+        LOGGER.warning(f"Output directory {output_dir} does not exist. Creating.")
+        output_dir.mkdir(parents=True)
+
+    AZURE_PROCESSOR_KEY = os.environ.get("AZURE_PROCESSOR_KEY")
```

Comment:
i'm not sure I understand why? this way the code only tries to access the environment if `__name__ == "__main__"` which would allow for importing other methods without having to access the environment

Reply:
https://peps.python.org/pep-0008/#constants